lib/src/isolate_manager/isolate_manager_interface.dart

```diff
@@ -0,0 +1,107 @@
+import 'dart:async';
+
+import 'package:dash_kit_network/dash_kit_network.dart';
+
+// This is an abstract class for managing isolates It defines the basic methods
+// needed to start, stop and send messages to an isolate. Flutter Web doesn't
+// support isolates so the `IsolateManager` class has a different implementation
+// for web.
+abstract class IsolateManagerInterface {
+  // `_isolateStatus` keeps track of the current status of the isolate.
+  IsolateStatus _isolateStatus = IsolateStatus.created;
+
+  // Method for starting the isolate. Should be called before sending messages.
+  // If `start()` is called multiple times, an assertion error is thrown.
+  Future<void> start() async {
```

Comment:
let's mark all public methdos with:

```
@mustCallSuper
```

as it done in Flutter for `StatefulWidget` for example

Reply:
Added for all methods.